### DISCUSSION TITLE
How can I update the BattleNet launcher?

### DISCUSSION THREAD

**Discussion by daviator** [moraroy/NonSteamLaunchers-On-Steam-Deck#75] [Q&A] [answered]
I tried to update the BattleNet launcher with the button/prompt once it is launched. But then it crashes... Is there another way to update the BattleNet launcher (automatically)?

**Comment by moraroy** [comment, marked as answer]
@daviator there is some YouTube videos.. but the script doesn't update anything for you... blizzard updates that launcher just like it would normally.. so the issues you have is with them.

**Reply by daviator** [reply]
Thanks, then I will look for my answer on Youtube. (GoG and Epic launchers update just fine btw)

**Comment by moraroy** [comment]
@daviator im pasting this here for you... you can add those little commands for each game you want, so i tested this for battle.net and it was just as you guys have said in order for the correct "games" to launch. it doesnt display any error messages but the update symbol wont go away but it isnt an issue, here is what the shortcut should look like this goes for seperate app ids or if you kept them in one. i used both the new executive launch command and the legacy one. heres an example of my Overwatch 2

Target:
"/home/deck/.local/share/Steam/steamapps/compatdata/NonSteamLaunchers/pfx/drive_c/Program Files (x86)/Battle.net/Battle.net.exe" --exec="launch Pro"

Start in:
"/home/deck/.local/share/Steam/steamapps/compatdata/NonSteamLaunchers/pfx/drive_c/Program Files (x86)/Battle.net/"

Launch Options:
STEAM_COMPAT_DATA_PATH="/home/deck/.local/share/Steam/steamapps/compatdata/NonSteamLaunchers/" %command% "battlenet://Pro"

i learned this form here: https://steamcommunity.com/sharedfiles/filedetails/?id=1113049716

**Reply by daviator** [reply]
Thanks, I replaced the Pro with Fren and now it starts Diablo 4. Brilliant!
The update is still bugged for me with Proton GE 8.6...
